src/quickgui/CMakeLists.txt

```diff
@@ -85,10 +85,6 @@ add_library(qgis_quick ${LIBRARY_TYPE}
     ${QGIS_QUICK_GUI_MOC_HDRS}
     ${QGIS_QUICK_GUI_MOC_SRCS}
     ${QGIS_QUICK_GUI_HDRS})
-
-# require c++17
-target_compile_features(qgis_quick PRIVATE cxx_std_17)
```

Comment:
this should be kept. if we cannot compile on android/ios, we should if-def it for those platforms